signing_service/signing_service/signing_service.py

```diff
@@ -135,6 +135,37 @@ def _validate_arguments(self):
 
 
 def _parse_arguments() -> argparse.Namespace:
+    def secret_provider_factory(read_command_line=False, env_variable_name=None, use_file=False, base64_convert=False):
+        def wrapper(**kwargs):
+            return SecretProvider(read_command_line, env_variable_name, use_file, base64_convert, **kwargs)
+        return wrapper
+
+    class SecretProvider(argparse.Action):
+        def __init__(self, read_command_line, env_variable_name, use_file, base64_convert, option_strings, dest):
+            self.read_command_line = read_command_line
+            self.env_variable_name = env_variable_name
+            self.use_file = use_file
+            self.base64_convert = base64_convert
+            if self.env_variable_name is not None:
+                command_line_arguments = 0
+            else:
+                command_line_arguments = None
+            super(SecretProvider, self).__init__(option_strings=option_strings, dest=dest, nargs=command_line_arguments)
+
+        def __call__(self, parser, namespace, values, option_string=None):
+            if values is not None and self.use_file:
+                with open(values) as file:
+                    self.const = file.read()
+            elif values is not None and self.read_command_line:
+                self.const = values
+            elif self.env_variable_name is not None:
+                self.const = os.environ.get(self.env_variable_name)
+            else:
+                assert False
+            if self.base64_convert and isinstance(self.const, str):
+                self.const = b64decode(self.const)
```

Comment:
What if `self.base64_convert` but not `isinstance(self.const, str)`? Should anything other than `str` be even allowed in such a case?

Reply:
It shouldn't. We expect that only string that is convertible from base64 to bytes will be passed here as `self.const`.

Reply:
And can user specify anything that will not be converted to string?
- If not, add an `assert`.
- If yes, this should be treated as any other error in parameters.

Reply:
I'd use the first one. But you could make it more concise:

``` python
if self.base64_convert:
    assert isinstance(self.constr, str)
    self.const = b64decode(self.const)
```